src/main/java/org/ballerinalang/command/util/ToolUtil.java

```diff
@@ -555,12 +571,24 @@ public static boolean downloadDistribution(PrintStream printStream, String distr
         }
     }
 
-    private static void downloadAndSetupDist(PrintStream printStream, HttpURLConnection conn,
+    private static void downloadDistributionZip(PrintStream printStream, HttpURLConnection conn,
                                              String distribution) {
         try {
-            String distPath = getDistributionsPath();
             String zipFileLocation = getDistributionsPath() + File.separator + distribution + ".zip";
             downloadFile(conn, zipFileLocation, distribution, printStream);
+        } finally {
+            conn.disconnect();
+        }
+    }
+
+    public static void setupDistribution (String distribution, String dependency) {
+        String distPath = getDistributionsPath();
+        String zipFileLocation = getDistributionsPath() + File.separator + distribution + ".zip";
+        if (!ToolUtil.checkDependencyAvailable(dependency)) {
+            new File(zipFileLocation).delete();
+            throw ErrorUtil.createCommandException("dependency '" + dependency + "' is not available locally. Please " +
```

Comment:
The required dependency is not available locally. Please try reinstalling the distribution.